src/main/java/pl/psnc/dei/ui/components/batches/ComplexBatchImportComponent.java

```diff
@@ -0,0 +1,208 @@
+import com.vaadin.flow.component.orderedlayout.HorizontalLayout;
+import com.vaadin.flow.component.orderedlayout.VerticalLayout;
+import com.vaadin.flow.component.select.Select;
+import com.vaadin.flow.component.textfield.TextField;
+import com.vaadin.flow.component.upload.Upload;
+import com.vaadin.flow.component.upload.receivers.MemoryBuffer;
+import com.vaadin.flow.data.provider.ListDataProvider;
+import com.vaadin.flow.dom.DomEvent;
+import com.vaadin.flow.dom.DomEventListener;
+import pl.psnc.dei.exception.NotFoundException;
+import pl.psnc.dei.model.Aggregator;
+import pl.psnc.dei.model.DAO.ProjectsRepository;
+import pl.psnc.dei.model.Dataset;
+import pl.psnc.dei.model.Project;
+import pl.psnc.dei.service.BatchService;
+import pl.psnc.dei.ui.components.CommonComponentsFactory;
+import pl.psnc.dei.util.ImportNameCreatorUtil;
+
+import java.io.IOException;
+import java.util.List;
+
+public class ComplexBatchImportComponent extends VerticalLayout {
+    private final ProjectsRepository projectsRepository;
+    private final BatchService batchService;
+
+    private Select<Project> projectSelect;
+    private Select<Aggregator> aggregatorSelect;
+    private Select<Dataset> datasetSelect;
+    private Button fileSelectButton;
+    private Button importButton;
+    private TextField nameTextFiled;
+
+    private Upload file;
+    private MemoryBuffer memoryBuffer;
+
+    public ComplexBatchImportComponent(ProjectsRepository projectsRepository, BatchService batchService) {
+        this.projectsRepository = projectsRepository;
+        this.batchService = batchService;
+        this.init();
+    }
+
+    private void init() {
+        this.initSelects();
+        this.initUploads();
+        this.setupLayout();
+    }
+
+    private void setupLayout() {
+        Div container = new Div();
+
+        HorizontalLayout selects = new HorizontalLayout();
+        selects.add(this.aggregatorSelect);
+        selects.add(this.projectSelect);
+        selects.add(this.datasetSelect);
+        container.add(selects);
+
+        HorizontalLayout importName = new HorizontalLayout();
+        importName.addClassName("vertically-centered-row");
+        importName.add(this.nameTextFiled);
+        container.add(importName);
+
+        HorizontalLayout uploads = new HorizontalLayout();
+        uploads.addClassName("vertically-centered-row");
+        uploads.add(this.file);
+        container.add(uploads);
+
+        add(container);
+
+        add(this.importButton);
+    }
+
+    private void initSelects() {
+        this.prepareProjectSelect();
+        this.prepareDatasetSelect();
+        this.prepareAggregatorSelect();
+    }
+
+    private void prepareAggregatorSelect() {
+        this.aggregatorSelect = CommonComponentsFactory.getAggregatorSelector();
+        this.aggregatorSelect.setValue(Aggregator.getById(0));
+        this.aggregatorSelect.setEmptySelectionAllowed(false);
+        this.aggregatorSelect.setValue(Aggregator.EUROPEANA);
```

Comment:
I'm so sorry for not noticing it previously, but there is duplicated `setValue` method call, please remove one of them if it's not a problem.

Reply:
Right, this in line 88 is invalid at all